hermit-macro/src/system.rs

```diff
@@ -59,13 +59,27 @@ fn parse_sig(sig: &Signature) -> Result<ParsedSig> {
 }
 
 fn validate_attrs(attrs: &[Attribute]) -> Result<()> {
+	let mut no_mangle_found = false;
 	for attr in attrs {
-		if !attr.path().is_ident("cfg") && !attr.path().is_ident("doc") {
+		if !attr.path().is_ident("cfg")
+			&& !attr.path().is_ident("doc")
+			&& !attr.path().is_ident("no_mangle")
+		{
 			bail!(
 				attr,
-				"#[system] functions may only have `#[doc]` and `#[cfg]` attributes"
+				"#[system] functions may only have `#[doc]`, `#[no_mangle]` and `#[cfg]` attributes"
 			);
 		}
+		if attr.path().is_ident("no_mangle") {
+			no_mangle_found = true;
+		}
+	}
+
+	if !no_mangle_found {
+		bail!(
+			attrs.first(),
+			"#[system] functions must have `#[no_mangle]` attribute"
+		);
 	}
```

Comment:
Not sure where if the best place to document the system calls would be here or in hermit-abi or somewhere else entirely. It might be a good idea eventually, but this is out of scope for this PR.